docs/cookbook/prompt_engineering/text_based/self_ask.md

```diff
@@ -0,0 +1,248 @@
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Which country is Ineabelle Diaz a citizen of?
+            Intermediate answer: Ineabelle Diaz is from Peurto Rico, which is in the United States of America.
+            Follow up: What is the current official currency in the United States of America?
+            Intermediate answer: The current official currency in the United States is the United States dollar.
+            So the final answer is: United States dollar.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="Where was the person who founded the American Institute of Public Opinion in 1935 born?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Who founded the American Institute of Public Opinion in 1935?
+            Intermediate answer: George Gallup.
+            Follow up: Where was George Gallup born?
+            Intermediate answer: George Gallup was born in Jefferson, Iowa.
+            So the final answer is: Jefferson.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="What language is used by the director of Tiffany Memorandum?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Who directed the movie called Tiffany Memorandum?
+            Intermediate answer: Sergio Grieco.
+            Follow up: What language is used by Sergio Grieco?
+            Intermediate answer: Sergio Grieco speaks Italian.
+            So the final answer is: Italian.
+            """
+        ),
+    ),
+    FewShotExample(
+        question="What is the sports team the person played for who scored the first touchdown in Superbowl 1?",
+        answer=inspect.cleandoc(
+            """
+            Are follow up questions needed here: Yes.
+            Follow up: Which player scored the first touchdown in Superbowl 1?
+            Intermediate answer: Max McGee.
+            Follow up: Which sports team did Max McGee play for?
+            Intermediate answer: Max McGee played for the Green Bay Packers.
+            So the final answer is: Green Bay Packers.
+            """
+        ),
+    ),
+]
+```
+
+Finally, we can use our Self-Ask implementation:
+
+```python
+response = self_ask_query(
+    query="The birth country of Jayantha Ketagoda left the British Empire when?",
+    examples=few_shot_examples,
+)
+print(response.content)
+# > Are follow up questions needed here: Yes.
+#   Follow up: What is the birth country of Jayantha Ketagoda?
+#   Intermediate answer: Jayantha Ketagoda is from Sri Lanka.
+#   Follow up: When did Sri Lanka leave the British Empire?
+#   Intermediate answer: Sri Lanka, formerly known as Ceylon, gained independence from the British Empire on February 4, 1948.
+#   So the final answer is: February 4, 1948.
+```
+
+## Enhancing Self-Ask Implementation
```

Comment:
per your slack comment, just making sure that it's fine for me to only do up to this part for the rest of text based prompt engineering techniques

Reply:
I think I'll just take all of the text-based ones so I can do the entire thing. Ideally any recipe you take you'll do the whole thing.